src/server.rs

```diff
@@ -89,6 +90,55 @@ pub extern "C" fn rustls_server_config_builder_from_config(
     }
 }
 
+/// Set the TLS protocol versions to use when negotiating a TLS session.
+///
+/// `tls_version` is the version of the protocol, as defined in rfc8446,
+/// ch. 4.2.1 and end of ch. 5.1. Some values are defined in
+/// `rustls_tls_version` for convenience.
+///
+/// `versions` will only be used during the call and the application retains
+/// ownership. `len` is the number of consecutive `ui16` pointed to by `versions`.
+#[no_mangle]
+pub extern "C" fn rustls_server_config_builder_set_versions(
+    builder: *mut rustls_server_config_builder,
+    tls_versions: *const u16,
+    len: size_t,
+) -> rustls_result {
+    ffi_panic_boundary! {
+        let config: &mut ServerConfig = try_ref_from_ptr!(builder, &mut ServerConfig);
+        config.versions.clear();
+        unsafe {
+            let x: &[u16] = slice::from_raw_parts(tls_versions, len);
+            for i in x {
+                match rustls_tls_version_from_u16(*i) {
+                    Some(pversion) => config.versions.push(pversion),
+                    None => return rustls_result::AlertIllegalParameter
+                }
+            }
+        }
+        rustls_result::Ok
+    }
+}
+
+/// Find out if the given TLS protocol version is suported in the
+/// `rustls_server_config` that is being built.
+/// `tls_version` is the version of the protocol, as defined in rfc8446,
+///  ch. 4.2.1 and end of ch. 5.1. Some values are defined in
+/// `rustls_tls_version` for convenience.
+#[no_mangle]
+pub extern "C" fn rustls_server_config_builder_supports_version(
+    builder: *mut rustls_server_config_builder,
+    tls_version: u16,
+) -> bool {
```

Comment:
Out of curiosity, is this something you need or anticipate needing in mod_tls?

Reply:
I'll remove this again. I was originally thinking that this method would be sufficient for `mod_tls`. I have to issue open at https://github.com/ctz/rustls/issues/577 to allow the API user to detect and iterate the supported versions.

It might seem far fetch to prepare for future protocol versions, but I guess people felt the same at the time `SSLv3` was hot.

Reply:
Removed the function again.